activerecord/lib/active_record/connection_adapters/schema_cache.rb

```diff
@@ -79,9 +79,14 @@ def columns(table_name)
       # Get the columns for a table as a hash, key is the column name
       # value is the column object.
       def columns_hash(table_name)
-        @columns_hash[table_name] ||= Hash[columns(table_name).map { |col|
-          [col.name, col]
-        }]
+        @columns_hash[table_name] ||= begin
+          if table_name.blank?
+            raise ActiveRecord::NoTableError, "columns_hash is not available for a model with no table"
```

Comment:
@gmcgibbon ended up moving the raising of the exception here, since this is actually the point at which the exception is raised.